### ISSUE TITLE
Segmentation fault in jv_unicode.c

### ISSUE THREAD

**Issue by moraxy** [jqlang/jq#1175]
For some reason compiling from source doesn't work for me, so this is tested against the latest source release. For what it's worth, jqplay.org also returns a `signal: segmentation fault`.

```
jq -n '""*"*9+<80>*"*9+8868*"*9+<80>*"*9+8868'
```

gdb:

```
Program received signal SIGSEGV, Segmentation fault.
0x00000000004a5d68 in jvp_utf8_next (codepoint_ret=<synthetic pointer>, end=<optimized out>, end@entry=0x745490 <error: Cannot access memory at address 0x745490>,
    in=0x73f000 <error: Cannot access memory at address 0x73f000>,
    in@entry=0x733f70 "*9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>**9+<80>*"...)
   at jv_unicode.c:12
12        unsigned char first = (unsigned char)in[0];
```

**Comment by pkoppstein** [comment]
On a Mac, there is no problem using jq 1.4, jq 1.5, or the "master" version of jq:
```

$ /usr/local/bin/jq --version
jq-1.4

$ /usr/local/bin/jq -n '""*"*9+<80>*"*9+8868*"*9+<80>*"*9+8868'
jq: error: string and string cannot be multiplied


$ ~/jq-1.5/jq --version
jq-1.5

$ ~/jq-1.5/jq -n '""*"*9+<80>*"*9+8868*"*9+<80>*"*9+8868'
jq: error (at <unknown>): string ("") and string ("*9+<80>*") cannot be multiplied


$ jq --version
jq-1.5rc2-150-g1740fd0
$ jq -n '""*"*9+<80>*"*9+8868*"*9+<80>*"*9+8868'
jq: error (at <unknown>): string ("") and string ("*9+<80>*") cannot be multiplied


```

**Comment by moraxy** [comment]
@pkoppstein only tested it on linux, maybe it's the terminal handling unicode differently.
Does
```
jq -n '"*"*0+99998*"\\"*8/868'
```
segfault for you?

**Comment by wtlangford** [comment]
@moraxy, can we also get the output from `uname -srvmpio` (That's `uname
-a` without your machine's nodename) on your machine?

**Comment by moraxy** [comment]
`Linux 4.4.0-24-generic #43-Ubuntu SMP Wed Jun 8 19:27:37 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by pkoppstein** [comment]
@moraxy - On my Mac, the shorter test case passes swimmingly.

However, I can confirm the segfault on a Linux machine:

```
-bash-4.1$ uname -srvmpio
Linux 2.6.32-642.1.1.el6.x86_64 #1 SMP Tue May 31 23:03:57 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux
-bash-4.1$  jq -n '"*"*0+99998*"\\"*8/868'
Segmentation fault
-bash-4.1$ jq --version
jq-1.5

```

**Comment by itchyny** [comment]
I can reproduce the issue with jq 1.6. This is likely the instance of same issue of #2117 and already fixed by #2118, so closing.
